coffee-module/coffee-module-redispubsub/src/main/java/hu/icellmobilsoft/coffee/module/redispubsub/PubSubSink.java

```diff
@@ -0,0 +1,109 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.module.redispubsub;
+
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+import hu.icellmobilsoft.coffee.dto.common.LogConstants;
+import hu.icellmobilsoft.coffee.module.redis.annotation.RedisConnection;
+import hu.icellmobilsoft.coffee.module.redis.manager.RedisManager;
+import hu.icellmobilsoft.coffee.module.redispubsub.bundle.PubSubMessage;
+import hu.icellmobilsoft.coffee.se.logging.Logger;
+import hu.icellmobilsoft.coffee.se.logging.mdc.MDC;
+import hu.icellmobilsoft.coffee.tool.gson.JsonUtil;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Sink for redis pubsub. Used by connector to consume emitted messages and publish them to redis. Messages can be emitted via
+ * {@link org.eclipse.microprofile.reactive.messaging.Outgoing} annotation or {@link org.eclipse.microprofile.reactive.messaging.Emitter#send} method
+ *
+ * @author mark.petrenyi
+ * @since 1.1.0
+ */
+public class PubSubSink {
+    private final PubSubConnectorOutgoingConfiguration outConfig;
+    private final Logger log = Logger.getLogger(PubSubSink.class);
+
+    /**
+     * Instantiates a new PubSubSink.
+     *
+     * @param outConfig
+     *            the outgoing configuration
+     */
+    public PubSubSink(PubSubConnectorOutgoingConfiguration outConfig) {
+        this.outConfig = outConfig;
+    }
+
+    /**
+     * Publish message to redis pub sub. Always sends {@link PubSubMessage} as json to redis channel
+     *
+     * @param msg
+     *            the mp reactive streams message to be sent
+     */
+    public void publishMessage(Message<?> msg) {
+        log.trace(">> publishMessage:[{0}]", msg);
+        String messagePayload = getPubSubMessage(msg);
+        String channel = outConfig.getPubSubChannel().orElseGet(outConfig::getChannel);
+        Instance<RedisManager> redisManagerInstance = CDI.current().select(RedisManager.class,
+                new RedisConnection.Literal(outConfig.getConnectionKey(), outConfig.getPoolKey()));
+        RedisManager redisManager = null;
+        try {
+            redisManager = redisManagerInstance.get();
+            Optional<Long> published = redisManager.runWithConnection(Jedis::publish, "publish", channel, messagePayload);
+            log.trace("Message published to [{0}] clients!", published.orElse(0L));
+            // redis pub/sub nem kezel ack/nack-ot, de az ide érkező message jöhet olyan forrásból,
+            // ahol kell, egyelőre ha sikeres a kiküldés ackolunk, ha nem akkor nack, később ezen lehet finomítani.
```

Comment:
es ha published != 0 lenne az ACK?

Reply:
Hát mindenképp kell a reactive message-re ack vagy nack. Szerintem jobb ha 0-nál is ack-olunk, mert részünkről sikerült a kiküldés, csak nem volt aki hallja.